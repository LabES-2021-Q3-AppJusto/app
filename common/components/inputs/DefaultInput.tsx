import React, { ReactNode, useCallback, useRef } from 'react';
import { Text, TextInput, TextInputProps, TouchableWithoutFeedback, View } from 'react-native';
import { borders, colors, texts } from '../../styles';
import PaddedView from '../containers/PaddedView';

export interface DefaultInputProps extends TextInputProps {
  title?: string;
  children?: ReactNode;
}

export default React.forwardRef(
  ({ title, children, editable = true, style, ...props }: DefaultInputProps, externalRef) => {
    const internalRef = useRef<TextInput>(null);
    const ref = (externalRef as React.RefObject<TextInput>) || internalRef;
    const focus = useCallback(() => {
      if (!ref.current) return null;
      // if (!ref.current.isFocused()) {
      ref.current.focus();
      // }
    }, [ref]);
    return (
      <PaddedView
        half
        style={[
          {
            ...borders.default,
            backgroundColor: 'white',
            flexDirection: 'row',
            justifyContent: 'space-between',
            alignItems: 'center',
          },
          style,
        ]}
      >
        <View style={{ flex: 1 }}>
          {title && (
            <TouchableWithoutFeedback onPress={focus}>
              <Text
                style={{
                  ...texts.xs,
                  color: colors.green600,
                  width: '100%',
                }}
              >
                {title}
              </Text>
            </TouchableWithoutFeedback>
          )}
          <TextInput
            ref={ref}
            style={{
              ...texts.md,
              color: editable ? colors.grey700 : colors.grey500,
              width: '100%',
            }}
            editable={editable}
            {...props}
          />
        </View>
        {children}
      </PaddedView>
    );
  }
);
