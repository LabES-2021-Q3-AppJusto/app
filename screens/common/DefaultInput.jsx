import React from 'react';
import { View, Text, TextInput, StyleSheet } from 'react-native';

import { borders, texts, colors } from './styles';

export default ({ title, children, ...props }) => (
  <View style={styles.container}>
    <View style={styles.inputContainer}>
      <Text style={styles.text}>{title}</Text>
      <TextInput {...props} />
    </View>
    {children}
  </View>
);

const styles = StyleSheet.create({
  container: {
    width: '100%',
    height: 61,
    ...borders.default,
    backgroundColor: 'white',
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    padding: 12,
  },
  inputContainer: {},
  text: {
    ...texts.default,
    color: colors.darkGreen,
    fontSize: 13,
    lineHeight: 16,
    paddingVertical: 2,
  },
});
