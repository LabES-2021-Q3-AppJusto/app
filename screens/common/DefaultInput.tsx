import React, { ReactNode } from 'react';
import { View, Text, TextInput, StyleSheet, TextInputProps } from 'react-native';

import { borders, texts, colors } from './styles';

export interface Props extends TextInputProps {
  title: String;
  children?: ReactNode;
}

export default ({ title, children, style: externalStyle, ...props }: Props) => (
  <View style={[style.container, externalStyle]}>
    <View>
      <Text style={style.text}>{title}</Text>
      <TextInput {...props} />
    </View>
    {children}
  </View>
);

const style = StyleSheet.create({
  container: {
    width: '100%',
    height: 60,
    ...borders.default,
    backgroundColor: 'white',
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    padding: 12,
  },
  text: {
    ...texts.default,
    color: colors.darkGreen,
    fontSize: 13,
    lineHeight: 16,
    paddingVertical: 2,
  },
});
