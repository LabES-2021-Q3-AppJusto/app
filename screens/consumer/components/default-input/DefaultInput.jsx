import React from 'react';
import { View, Text, TextInput, StyleSheet } from 'react-native';

import RegularButton from '../regular-button/RegularButton';
import { t } from '../../../../strings';

const DeafaultInput = ({ actionButton, title, placeholder, onChange, value, ...props }) => (
  <View style={styles.container}>
    <View style={styles.inputContainer}>
      <Text style={styles.text}>{title}</Text>
      <TextInput placeholder={placeholder} onChange={onChange} value={value} {...props} />
    </View>
    {actionButton && <RegularButton>{t('enter')}</RegularButton>}
  </View>
);

const styles = StyleSheet.create({
  container: {
    width: '100%',
    height: 61,
    borderRadius: 8,
    borderStyle: 'solid',
    borderColor: '#C8D7CB',
    borderWidth: 1,
    backgroundColor: 'white',
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    padding: 12,
  },
  inputContainer: {},
  text: {
    fontSize: 13,
    lineHeight: 16,
    color: '#63B745',
    paddingVertical: 2,
  },
});

export default DeafaultInput;
