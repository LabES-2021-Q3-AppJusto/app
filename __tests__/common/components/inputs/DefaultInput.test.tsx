import { fireEvent, render, waitFor } from '@testing-library/react-native';
import React, { useCallback, useState } from 'react';
import { Text } from 'react-native';
import DefaultInput from '../../../../common/components/inputs/DefaultInput';

const DefaultInputExample: React.FC = () => {
  const [text, setText] = useState('');
  const [inputError, setInputError] = useState('');
  const [trailing, setTrailing] = useState<React.ReactNode | undefined>(undefined);

  const onChangeText = useCallback(
    (text: string) => {
      setText(text);

      if (text === 'error') {
        setInputError('Input error');
      } else {
        setInputError('');
      }

      setTrailing(<Text>bla</Text>);
    },
    [setText, setInputError, setTrailing]
  );

  return (
    <DefaultInput
      title="Example"
      value={text}
      onChangeText={onChangeText}
      trailing={trailing}
      errorMessage={inputError}
    />
  );
};

it('Should render input properly', async () => {
  const { getByTestId, queryByTestId, toJSON } = render(<DefaultInputExample />);

  const inputText = 'example';

  const input = getByTestId('default-input-input');
  fireEvent.changeText(input, inputText);

  await waitFor(() => expect(queryByTestId('default-input-error-message')).toBeFalsy());

  expect(toJSON()).toMatchSnapshot();
});

it('Should display error message', async () => {
  const { getByTestId, queryByTestId, toJSON } = render(<DefaultInputExample />);

  const inputText = 'error';

  const input = getByTestId('default-input-input');
  fireEvent.changeText(input, inputText);

  await waitFor(() => expect(queryByTestId('default-input-error-message')).toBeTruthy());

  expect(toJSON()).toMatchSnapshot();
});
