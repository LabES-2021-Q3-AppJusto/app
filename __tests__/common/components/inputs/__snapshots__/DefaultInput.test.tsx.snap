// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should display error message 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "column",
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "paddingVertical": 8,
        },
        Object {
          "paddingHorizontal": 8,
        },
        Array [
          Object {
            "alignItems": "flex-start",
            "backgroundColor": "white",
            "borderColor": "#DC3545",
            "borderRadius": 8,
            "borderStyle": "solid",
            "borderWidth": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
          },
        ],
      ]
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <Text
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "color": "#4EA031",
            "fontFamily": "BarlowMedium",
            "fontSize": 13,
            "width": "100%",
          }
        }
        testID="default-input-title"
      >
        Example
      </Text>
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            editable={true}
            onChangeText={[Function]}
            rejectResponderTermination={true}
            style={
              Object {
                "color": "#697667",
                "fontFamily": "BarlowMedium",
                "fontSize": 16,
                "width": "100%",
              }
            }
            testID="default-input-input"
            underlineColorAndroid="transparent"
            value="error"
          />
        </View>
        <Text>
          bla
        </Text>
      </View>
    </View>
  </View>
  <Text
    style={
      Object {
        "color": "#DC3545",
        "fontFamily": "BarlowMedium",
        "fontSize": 13,
      }
    }
    testID="default-input-error-message"
  >
    Input error
  </Text>
</View>
`;

exports[`Should render input properly 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "column",
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "paddingVertical": 8,
        },
        Object {
          "paddingHorizontal": 8,
        },
        Array [
          Object {
            "alignItems": "flex-start",
            "backgroundColor": "white",
            "borderColor": "#C8D7CB",
            "borderRadius": 8,
            "borderStyle": "solid",
            "borderWidth": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
          },
        ],
      ]
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <Text
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "color": "#4EA031",
            "fontFamily": "BarlowMedium",
            "fontSize": 13,
            "width": "100%",
          }
        }
        testID="default-input-title"
      >
        Example
      </Text>
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            editable={true}
            onChangeText={[Function]}
            rejectResponderTermination={true}
            style={
              Object {
                "color": "#697667",
                "fontFamily": "BarlowMedium",
                "fontSize": 16,
                "width": "100%",
              }
            }
            testID="default-input-input"
            underlineColorAndroid="transparent"
            value="example"
          />
        </View>
        <Text>
          bla
        </Text>
      </View>
    </View>
  </View>
</View>
`;
